Form/ChoosePaymentMethodType.php

```diff
@@ -92,15 +92,18 @@ protected function buildChoiceList(FormBuilderInterface $builder, array $options
             $label = 'form.label.'.$method;
 
             if (Legacy::formChoicesAsValues()) {
-                $options['choices'][$method] = $label;
-            } else {
                 $options['choices'][$label] = $method;
+                if (Legacy::needsChoicesAsValuesOption()) {
+                    $options['choices_as_values'] = true;
```

Comment:
> this changes the behaviour

The option is change, but the contents of the array is changed too (it is flipped). So the result is the same, w/o the deprecation error